### DISCUSSION TITLE
Installing uno-check and the uno VS Plugin in an Offline environment

### DISCUSSION THREAD

**Discussion by alex6851** [unoplatform/uno#17597] [Q&A] [answered]
Does anyone know if uno-check and/or the uno VS plugin (and be able to create uno projects) can run in an offline environment?

I have tried running uno-check in an offline environment and i get an error as it's trying to reach out to raw.githubusercontent.com.

I have tried creating an uno solution/project in an online environment and then copying over the solution and nuget packages from the online environment to the offline environment and i couldn't get it to work, but I might be missing some steps.

**Comment by MartinZikmund** [comment, marked as answer]
Not for `uno-check` as it depends on various external sources to check for up-to-date versions and to fetch installation packages as required.

For uno templates that you pre-install you technically don't need to be connected, but you must make sure to have the required NuGet packages with the right versions already restored in NuGet package cache - so it means you would have to create the uno app from template with the required preset at least once on the device while connected (with the same exact version of Uno.Sdk). The second time you would create the same preset it should pass even without connection
